pkg/utils/utils.go

```diff
@@ -177,121 +182,49 @@ func InitInformerMap() {
 
 // GetObjectMetaData returns metadata of the given object
 func GetObjectMetaData(obj interface{}) metaV1.ObjectMeta {
-
-	var objectMeta metaV1.ObjectMeta
-
-	switch object := obj.(type) {
-	case *coreV1.Event:
-		objectMeta = object.ObjectMeta
-		// pass InvolvedObject`s annotations into Event`s annotations
-		// for filtering event objects based on InvolvedObject`s annotations
+	unstructuredObject := obj.(*unstructured.Unstructured).DeepCopy()
```

Comment:
We should check if assertion is possible first, otherwise, this would cause panic
```suggestion
	unstructuredObject, ok := obj.(*unstructured.Unstructured)
	if !ok {
	        // log and return
	}
```

Reply:
Not resolved yet.